### DISCUSSION TITLE
type hinting with Union

### DISCUSSION THREAD

**Discussion by caffeinism** [fastapi/fastapi#6924] [Questions] [answered]
I started writing an issue because I had a question about the type hint and behavior of fastapi. I recently encountered fastapi and made several toy examples, and I had that question, but I couldn't solve it by searching.

The problem with the behavior of my code is how to interpret the type hinted parameter as a union passed to the request body. I have specified the type as Union[int, float, str, ...] to receive various types of data. But when I passed 1.1, I found it interpreted as 1.

So, after doing some experiments, I found that among the types specified by Union, the type was changed to the first interpretable type.

For example
```
input: 1

type: Union[int,float,str]
input actually: 1

type: Union[float,str,int]
input actually: 1.0

type: Union[str,int,float]
input actually: '1'
```
Is there any way to solve this?

Thanks for reading this issue.

**Comment by Kludex** [comment, marked as answer]
It tries to match in order. 

Check [Strict Types](https://pydantic-docs.helpmanual.io/usage/types/#strict-types) for strictness.

**Comment by caffeinism** [comment]
@Kludex 

There was such a document. Thank you very much for your answer.

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! :clap: :bow: 

Thanks for reporting back and closing the issue @caffeinism  :+1:
